### PR TITLE
Log timing information for processes in server jobs

### PR DESCRIPTION
**Pull request by nwalters512**
As we prepare to deploy syncing changes, this should give us a little more information.

Screenshot 2024-03-05 at 11 18 50

### CODE REVIEW

File: apps/prairielearn/src/lib/server-jobs.ts [hunk 1]
```diff
@@ -95,22 +95,30 @@ class ServerJobImpl implements ServerJob, ServerJobExecutor {
     this.addToOutput(chalk.blueBright(`Command: ${file} ${args.join(' ')}\n`));
     this.addToOutput(chalk.blueBright(`Working directory: ${options.cwd}\n`));
 
+    const start = Date.now();
```
Comment (mwest1066) [inline apps/prairielearn/src/lib/server-jobs.ts:98]: Nit: are we supposed to use [`performance.now()`](https://developer.mozilla.org/en-US/docs/Web/API/Performance/now) for timing?

**Review by mwest1066 — APPROVED** [review]
I'm now also curious about how long some of the JS steps take (e.g., "Sync to database"). Maybe we should also manually add timing output? I'm wondering about a helper in `job` so we could call `job.resetTimer()` before a command and `job.resetTimerAndPrintElapsedTime()` afterwards (hopefully with better names)?

File: apps/prairielearn/src/lib/server-jobs.ts (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/src/lib/server-jobs.ts:98]: We probably should, yes!

Comment (nwalters512) [reply apps/prairielearn/src/lib/server-jobs.ts]: Done in 7640839.

**Comment by nwalters512** [comment]
@mwest1066 I have an idea for an API for tracking execution time of regular code; I'll open a separate PR for that.

**Comment by mwest1066** [comment]
> @mwest1066 I have an idea for an API for tracking execution time of regular code; I'll open a separate PR for that.

Excellent!

**Pull request merged**